arc-script/arc-script-test/compile/src/snapshots/hir@structs.arc.snap

```diff
@@ -5,7 +5,7 @@ input_file: arc-script-test/compile/src/tests/expect_pass/structs.arc
 
 ---
 fun test() -> unit {
-    let x_0: { a: { c: i32, b: i32 }, xyz: i32, d: { b: i32, c: i32 } } = { a: { c: 1, b: 5 }, xyz: 2, d: { b: 4, c: 2 } } in
+    let x_0: { a: { b: i32, c: i32 }, d: { b: i32, c: i32 }, xyz: i32 } = { a: { b: 5, c: 1 }, d: { b: 4, c: 2 }, xyz: 2 } in
```

Comment:
Is this safe to do? What if the code was:

```
fun foo(a: {c: i32, b: i32}, b: {b:i32, c:i32}) ->
    {a:{c:i32, b:i32}, xyz:i32, d:{b:i32, c:i32}} {
    let r = {a:a, xyz:get_counter(), d:get_counter()} in
    r
}
```

where `get_counter()` has a side effect? Have we made / should we make any promises about the evaluation order of initializers?

Reply:
Good point, fixed now!

Reply:
I also ran `cargo fmt` which changed some other files, sorry 👀